core/models/dataset.py

```diff
@@ -227,6 +238,79 @@ def publish(self, save=True):
         for data_declaration in self.data_declarations.all():
             data_declaration.publish_subentities()
 
+    @staticmethod
+    def get_notification_recipients():
+        """
+        Get distinct users that are local custodian of a dataset or a project.
+        """
+
+        return (
+            get_user_model()
+            .objects.filter(Q(datasets__isnull=False) | Q(project_set__isnull=False))
+            .distinct()
+        )
+
+    @classmethod
+    def make_notifications(cls, exec_date: datetime.date):
+        recipients = cls.get_notification_recipients()
+        for user in recipients:
+            notification_setting: NotificationSetting = (
+                Dataset.get_notification_setting(user)
+            )
+            if not (
+                notification_setting.send_email or notification_setting.send_in_app
+            ):
+                continue
+            day_offset = timedelta(days=notification_setting.notification_offset)
+
+            # Considering users that are indirectly responsible for the dataset (through projects)
+            possible_datasets = set(user.datasets.all())
+            for project in user.project_set.all():
+                possible_datasets.update(list(project.datasets.all()))
```

Comment:
not sure the conversion to list is mandatory here, update should accept an iterable no?